src/app/entities/courses/directives/show-course-status/show-course-status.directive.ts

```diff
@@ -0,0 +1,35 @@
+import { Directive, HostBinding, Input, OnInit } from '@angular/core';
+
+@Directive({
+  selector: '[appShowCourseStatus]',
+})
+export class ShowCourseStatusDirective implements OnInit {
+  FOURTEEN_DAYS_IN_MILLISECONDS = 1000 * 60 * 60 * 24 * 14;
+  COLORS = {
+    BLUE: '#009ECC',
+    GREEN: '#67A300',
+    TRANSPARENT: 'transparent',
+  }
+
+  @HostBinding('style.border-color')
+  borderColor = this.COLORS.TRANSPARENT;
+
+  @Input('appShowCourseStatus') creationDate: Date;
```

Comment:
he is some inconsistency: status is not a date at all.
It may be a `AvailabilityStatusDirective` that accepts some date...

Reply:
here you can use `setter` approach
```typescript
@Input() set availabilityDate(date: Date) {
   this.borderColor = calculateBorderColor(date);
}
```
and `calculateBorderColor` in this case can become a simple helper function...

Reply:
Agree